From 5166ecc02a73f856d88815f5757ba5c6696d3360 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Thu, 13 Apr 2017 10:03:06 +0200
Subject: [PATCH 01/13] google: Do not ignore errors when rebuilding cache

If file cache is not loaded properly (e.g. because of cancellation),
it may lead to various problems (e.g. incomplete enumeration). Let's
return error in such case and rebuild the cache next time...

https://bugzilla.gnome.org/show_bug.cgi?id=781252
---
 daemon/gvfsbackendgoogle.c | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/daemon/gvfsbackendgoogle.c b/daemon/gvfsbackendgoogle.c
index 376c2309..7bed6084 100644
--- a/daemon/gvfsbackendgoogle.c
+++ b/daemon/gvfsbackendgoogle.c
@@ -557,18 +557,11 @@ rebuild_entries (GVfsBackendGoogle  *self,
       feed = gdata_documents_service_query_documents (self->service, query, cancellable, NULL, NULL, &local_error);
       if (local_error != NULL)
         {
-          if (succeeded_once)
-            {
-              g_warning ("Unable to query: %s", local_error->message);
-              g_error_free (local_error);
-            }
-          else
-            {
-              sanitize_error (&local_error);
-              g_propagate_error (error, local_error);
-            }
+          sanitize_error (&local_error);
+          g_propagate_error (error, local_error);
+          self->entries_stale = TRUE;
 
-          break;
+          goto out;
         }
 
       if (!succeeded_once)
@@ -598,6 +591,7 @@ rebuild_entries (GVfsBackendGoogle  *self,
 
   self->entries_stale = FALSE;
 
+ out:
   g_clear_object (&feed);
   g_clear_object (&query);
 }
-- 
2.15.1

